apps/emqx_durable_storage/src/emqx_ds_beamformer.erl

```diff
@@ -322,8 +319,11 @@ handle_call(
     _From,
     S = #s{pending_queue = PendingTab, wait_queue = WaitingTab, metrics_id = Metrics}
 ) ->
-    NQueued = ets:info(PendingTab, size) + ets:info(WaitingTab, size),
-    case NQueued >= S#s.pending_request_limit of
+    PQLen = ets:info(PendingTab, size),
+    WQLen = ets:info(WaitingTab, size),
```

Comment:
If this `#poll_req{}` call is hot path, calling `ets:info(_, size)` could be [very costly](https://github.com/emqx/emqx/pull/10573).